src/backend/distributed/transaction/relation_access_tracking.c

```diff
@@ -439,7 +439,10 @@ RecordRelationParallelDDLAccessForTask(Task *task)
 		lastRelationId = currentRelationId;
 	}
 
-	RecordParallelDDLAccess(RelationIdForShard(task->anchorShardId));
+	if (task->anchorShardId != INVALID_SHARD_ID)
```

Comment:
We should have a single task, and that's why `RecordParallelRelationAccessForTaskList()` should return early. So, do we really need this change?

Reply:
Might be I hit this code when I was creating separate tasks per node. Then again, if this value is set to invalid there is no reason to call the function below. Any concerns having this check here?